esp-hal/src/rtc_cntl/mod.rs

```diff
@@ -631,9 +631,9 @@ impl RtcClock {
         (100_000_000 * 1000 / period) as u16
     }
 
-    // TODO: implement for ESP32-C6
+    // TODO: implement for ESP32-C6, and H2
     #[cfg(not(any(esp32c6, esp32h2)))]
-    fn estimate_xtal_frequency() -> u32 {
+    pub fn estimate_xtal_frequency() -> u32 {
```

Comment:
I don't think this needs to be pub? It might need `pub crate` though, I can't remember the way the structure of that module works :D.